### PR TITLE
Add 1.17, 1.16.2, 1.16.7, 1.15.10

### PR DESCRIPTION
**Pull request by mble**
This commit updates the buildpack to make 1.17, 1.16.7, 1.16.6, and 1.15.10 available, and updates 1.16.7 as the default for 1.16 and 1.15.10 for 1.15.

It also updates the base URL for the archives to be https://dl.google.com/go, as the old `storage.googleapis.com` host does not contain new versions of Go. This also aligns with the official Docker image source.

It also updates the S3 bucket storage location which was out of sync with the name in lastpass.

Ref: https://groups.google.com/g/golang-announce/c/uHACNfXAZqk,
https://blog.golang.org/go1.17

### CODE REVIEW

**Comment by mble** [comment]
- [x] Upload assets to S3

File: files.json [hunk 1]
```diff
@@ -491,6 +495,18 @@
     "SHA": "3edc22f8332231c3ba8be246f184b736b8d28f06ce24f08168d8ecf052549769",
     "URL": "https://storage.googleapis.com/golang/go1.16.1.linux-amd64.tar.gz"
   },
+  "go1.16.2.linux-amd64.tar.gz": {
+    "SHA": "542e936b19542e62679766194364f45141fde55169db2d8d01046555ca9eb4b8",
+    "URL": "https://dl.google.com/go/go1.16.2.linux-amd64.tar.gz"
+  },
+  "go1.16.6.linux-amd64.tar.gz": {
```
Comment (edmorley) [inline files.json:514]: Should we add `go1.16.{3,4,5}` too?

File: data.json [hunk 2]
```diff
@@ -117,7 +119,8 @@
       "go1.14.2.linux-amd64.tar.gz",
       "go1.15.9.linux-amd64.tar.gz",
```
Comment (edmorley) [inline data.json]:
```suggestion
      "go1.15.10.linux-amd64.tar.gz",
```

**Comment by joshwlewis** [comment]
Synced to S3:

```
All files verified, syncing to s3

upload: ./go1.16.6.linux-amd64.tar.gz to s3://heroku-golang-prod/go1.16.6.linux-amd64.tar.gz
upload: ./go1.16.7.linux-amd64.tar.gz to s3://heroku-golang-prod/go1.16.7.linux-amd64.tar.gz
upload: ./go1.17.linux-amd64.tar.gz to s3://heroku-golang-prod/go1.17.linux-amd64.tar.gz
```

**Review by joshwlewis — APPROVED** [review]

File: files.json (hunk 1 above)
Comment (joshwlewis) [reply files.json:514]: Oddly, 1.16.{2,6,7} were the ones detected to be missing by the sync script. Will try the others.

Comment (joshwlewis) [reply files.json:514]: Added in 52dc633e0bdc1c9fdfbceddd81734156154a4e5a

**Comment by joshwlewis** [comment]
Three more synced to S3:

```
All files verified, syncing to s3

upload: ./go1.16.3.linux-amd64.tar.gz to s3://heroku-golang-prod/go1.16.3.linux-amd64.tar.gz
upload: ./go1.16.4.linux-amd64.tar.gz to s3://heroku-golang-prod/go1.16.4.linux-amd64.tar.gz
upload: ./go1.16.5.linux-amd64.tar.gz to s3://heroku-golang-prod/go1.16.5.linux-amd64.tar.gz
```

**Pull request merged**